docs/workflows/how-to/implement-steps.md

```diff
@@ -0,0 +1,166 @@
+---
+title: Implement steps
+sidebar_position: 1
+---
+
+# Implementing steps
+
+As explained on the [Pipeline Concepts](../explanations/pipeline.md#step) page, steps are linked from a pipeline using
+`code:link` property. Out of the box barnard59 supports JavaScript steps. The implementation of a step is a factory
+function which returns a stream or an async generator.
+
+This page presents three common ways to implement JavaScript steps.
+
+## Async generators
+
+The recommended way to implement a step is to use an async generator. Simply return an async generator from the factory.
+It must have a single argument which is the stream itself as an iterable. The generator can then yield values to the stream.
+
+```js
+import nanoid from 'nanoid'
+
+/**
+ * @this {import('barnard59-core').Context}
+ */
+export default function step() {
+  // 1. prepare the step
+  const id = nanoid()
+  const ex = this.rdf.namespace('http://example.org/')
+  
+  return async function* (stream) {
+    // 2. before first chunk
+    let total = 0
+    yield this.rdf.quad(rdf.blankNode(id), rdf.ns.rdf.type, ex.StepSummary)
```

Comment:
is it `this.rdf` or `this.env`?

Reply:
Good catch. Also missing some more of `this.`